app/apollo/resolvers/subscription.js

```diff
@@ -462,34 +492,59 @@ const subscriptionResolvers = {
         const kubeOwnerId = await models.User.getKubeOwnerId(context);
 
         const conditions = await getGroupConditionsIncludingEmpty(me, org_id, ACTIONS.READ, 'name', queryName, context);
-        logger.debug({req_id, user, org_id, conditions }, `${queryName} group conditions are...`);
+        logger.debug({req_id, user, org_id, conditions}, `${queryName} group conditions are...`);
 
         const subscription = await models.Subscription.findOne({ org_id, uuid, ...conditions }, {}).lean({ virtuals: true });
-        if(!subscription){
+        logger.info({req_id, user, org_id, uuid, name}, `${queryName} validating - found: ${!!subscription}`);
+
+        const identifiers = subscription ? [uuid, subscription.name] : [uuid];
+        await validAuth(me, org_id, ACTIONS.UPDATE, TYPES.SUBSCRIPTION, queryName, context, identifiers);
+        logger.info({req_id, user, org_id, uuid, name}, `${queryName} validating - authorized`);
+
+        // find the channel
+        const channel = await models.Channel.findOne({ org_id, uuid: channel_uuid });
+        logger.info({req_id, user, org_id, name, channel_uuid, version_uuid}, `${queryName} validating - found: ${!!channel}`);
+
+        const channelIdentifiers = channel ? [channel_uuid, channel.name] : [channel_uuid];
+        await validAuth(me, org_id, ACTIONS.READ, TYPES.CHANNEL, queryName, context, channelIdentifiers);
+        logger.info({req_id, user, org_id, name, channel_uuid, version_uuid}, `${queryName} validating - channel authorized`);
+
+        // find the groups
+        const foundGroups = await models.Group.find({
+          org_id,
+          $or: [
+            { name: { $in: groups } },
+            { uuid: { $in: groups } }
+          ]
+        }).lean({ virtuals: true });
+        logger.info({req_id, user, org_id, name, channel_uuid, version_uuid}, `${queryName} validating - found: ${foundGroups.length}`);
+
+        const allowedGroups = await filterResourcesToAllowed(me, org_id, ACTIONS.READ, TYPES.GROUP, foundGroups, context);
+        logger.info({req_id, user, org_id, name, channel_uuid, version_uuid}, `${queryName} validating - allowed: ${allowedGroups.length}`);
+
+        const groupNames = _.map(allowedGroups, group => group.name);
+
+        // Check for errors after validation
+        if(foundGroups.length < groups.length || allowedGroups.length < groups.length) {
+          throw new NotFoundError(context.req.t('One or more of the passed group uuids were not found'));
+        }
+        if( allowedGroups.length < foundGroups.length ) {
+          throw new NotFoundError(context.req.t('One or more of the passed group uuids were not found'));
+        }
+        if (!subscription) {
           throw new NotFoundError(context.req.t('Subscription { uuid: "{{uuid}}", org_id:{{org_id}} } not found.', {'uuid':uuid, 'org_id':org_id}), context);
         }
```

Comment:
The ordering of validation causes unnecessary load
- if there is no subscription found, there's no need to query channel or groups
- if there is no channel found, there is no need to query groups
- If the final list of allowed groups is less than requested input, either because they do not exist or the user is not authorized to them, the error is the same.  No need for separate error checks/errors.

```suggestion
        const identifiers = subscription ? [uuid, subscription.name] : [uuid];
        await validAuth(me, org_id, ACTIONS.UPDATE, TYPES.SUBSCRIPTION, queryName, context, identifiers);
        logger.info({req_id, user, org_id, uuid, name}, `${queryName} validating - authorized`);

        if (!subscription) {
          throw new NotFoundError(context.req.t('Subscription { uuid: "{{uuid}}", org_id:{{org_id}} } not found.', {'uuid':uuid, 'org_id':org_id}), context);
        }

        // find the channel
        const channel = await models.Channel.findOne({ org_id, uuid: channel_uuid });
        logger.info({req_id, user, org_id, name, channel_uuid, version_uuid}, `${queryName} validating - found: ${!!channel}`);

        const channelIdentifiers = channel ? [channel_uuid, channel.name] : [channel_uuid];
        await validAuth(me, org_id, ACTIONS.READ, TYPES.CHANNEL, queryName, context, channelIdentifiers);
        logger.info({req_id, user, org_id, name, channel_uuid, version_uuid}, `${queryName} validating - channel authorized`);

        if (!channel) {
          throw new NotFoundError(context.req.t('Channel { uuid: "{{uuid}}", org_id:{{org_id}} } not found.', {'uuid':channel_uuid, 'org_id':org_id}), context);
        }

        // find the groups
        const foundGroups = await models.Group.find({
          org_id,
          $or: [
            { name: { $in: groups } },
            { uuid: { $in: groups } }
          ]
        }).lean({ virtuals: true });
        logger.info({req_id, user, org_id, name, channel_uuid, version_uuid}, `${queryName} validating - found: ${foundGroups.length}`);

        const allowedGroups = await filterResourcesToAllowed(me, org_id, ACTIONS.READ, TYPES.GROUP, foundGroups, context);
        logger.info({req_id, user, org_id, name, channel_uuid, version_uuid}, `${queryName} validating - allowed: ${allowedGroups.length}`);

        if(allowedGroups.length < groups.length) {
          throw new NotFoundError(context.req.t('One or more of the passed groups were not found'));
        }

        const groupNames = _.map(allowedGroups, group => group.name);
```